src/components/vm/nics/nicAdd.tsx

```diff
@@ -73,275 +77,235 @@ function getRandomMac(vms: VM[]): string | undefined {
-
+                    portForward: (
+                        values.type == "user"
+                            ? dialogPortForwardsToInterface(values.portForwards)
+                            : null
+                    ),
                 },
-                this.state.permanent
+                values.permanent
             );
             domainGet({ connectionName: vm.connectionName, id: vm.id });
-            Dialogs.close();
         } catch (exc) {
-            this.dialogErrorSet(_("Network interface settings could not be saved"), String(exc));
+            throw DialogError.fromError(_("Network interface settings could not be saved"), exc);
         }
-        this.setState({ addVNicInProgress: false });
     }
 
-    render() {
-        const Dialogs = this.context;
-        const { idPrefix, vm } = this.props;
-
-        const defaultBody = (
-            <>
-                <Form onSubmit={e => e.preventDefault()} isHorizontal>
-                    <NetworkTypeAndSourceRow
-                        vm={vm}
-                        idPrefix={idPrefix}
-                        dialogValues={{ ...this.state, availableSources: this.props.availableSources }}
-                        onValueChanged={this.onBodyValueChanged}
-                    />
-
-                    <NetworkModelRow
-                        idPrefix={idPrefix}
-                        dialogValues={this.state}
-                        onValueChanged={this.onBodyValueChanged}
-                        osTypeArch={vm.arch}
-                        osTypeMachine={vm.emulatedMachine}
-                    />
-
-                    <NetworkMacRow
-                        idPrefix={idPrefix}
-                        dialogValues={this.state}
-                        onValueChanged={this.onValueChanged}
-                    />
-
-                    { domainIsRunning(vm.state) && vm.persistent &&
-                        <PermanentChange
-                            idPrefix={idPrefix}
-                            dialogValues={this.state}
-                            onValueChanged={this.onValueChanged}
-                        />
-                    }
-                </Form>
-                { this.state.networkType == "user" &&
-                    vm.capabilities.interfaceBackends.includes("passt") &&
-                    <Form>
-                        <br />
-                        <NetworkPortForwardsRow
-                            idPrefix={idPrefix}
-                            dialogValues={this.state}
-                            validation={this.state.validation}
-                            onValueChanged={this.onBodyValueChanged}
-                        />
-                    </Form>
+    const dlg = useDialogState(init, validate);
+
+    const defaultBody = (
+        <>
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2425-899c38e3-20251201-074921-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)